src/main/kotlin/de/livepoll/api/service/PollItemService.kt

```diff
@@ -219,36 +256,18 @@ class PollItemService {
     }
 
     fun updateQuizItem(pollItemId: Long, pollItem: QuizItemWithPositionDtoIn): QuizItemDtoOut {
+        if (pollItem.selectionOptions.size < 2) {
+            throw ResponseStatusException(
+                HttpStatus.BAD_REQUEST,
+                "Quiz item must include at least two items (correct selection option and at least one wrong selection option)"
+            )
+        }
+
         quizItemRepository.findById(pollItemId)
             .orElseThrow { ResponseStatusException(HttpStatus.NOT_FOUND, "Poll item not found") }
             .run {
-                val newAnswers = pollItem.answers.toMutableList()
-                val removeAnswer = mutableListOf<QuizItemAnswer>()
-                this.answers.forEach {
-                    if (it.answerCount != 0) {
-                        if (pollItem.answers.contains(it.selectionOption)) {
-                            newAnswers.remove(it.selectionOption)
-                            it.isCorrect = false
-                        } else {
-                            this.answers.remove(it)
-                        }
-                    } else {
-                        if (!pollItem.answers.contains(it.selectionOption)) {
-                            removeAnswer.add(it)
-                        }
-                    }
-                }
-                removeAnswer.forEach {
-                    this.answers.remove(it)
-                }
-                newAnswers.forEach {
-                    this.answers.add(QuizItemAnswer(0, this, it, false, 0))
-                }
-                val newCorrectOne = this.answers.find { it.selectionOption == pollItem.answers[0] }!!
-                if (newCorrectOne.answerCount == 0) {
-                    this.answers.find { it.selectionOption == pollItem.answers[0] }!!.isCorrect = true
-                }
-
+                updateAnswers(this, pollItem.selectionOptions)
+                this.answers[0].isCorrect = true // Make sure that first answer is correct one
```

Comment:
When I look at the updateAnswer method, I think we can't be sure that the first item here is the correct one. New added items are inserted at the end of the existing list, so that the wrong item could be marked as correct (if there is a new answer at the top of the list). Take a look at my old code, that is why I am looking for the first element of the update list again

Reply:
True, thanks for noticing, I've updated the logic and added checks for whether it is allowed to have a new correct item or not.